### PR TITLE
[ch21613] PD results - 'Effective and efficient programme management' component updates

### PR DESCRIPTION
**Pull request by laza88**
[ch21613] PD results - 'Effective and efficient programme management' component

### CODE REVIEW

File: intervention-results/effective-efficient-programme-mgmt/effective-efficient-programme-mgmt.ts [hunk 1]
```diff
@@ -88,21 +74,28 @@ export class EffectiveAndEfficientProgrammeManagement extends connect(getStore()
...
+          .showEdit=${this.canEditActivity}
+          @edit-item="${this.editItem}"
           .getChildRowTemplateMethod="${this.getChildRowTemplate.bind(this)}"
         >
         </etools-table>
       </etools-content-panel>
     `;
   }
 
+  private formattedData;
```
Comment (acory) [inline intervention-results/effective-efficient-programme-mgmt/effective-efficient-programme-mgmt.ts:89]: given it is used in template, should probably be a property

File: intervention-results/effective-efficient-programme-mgmt/effective-efficient-programme-mgmt.ts [hunk 2]
```diff
@@ -151,32 +145,65 @@ export class EffectiveAndEfficientProgrammeManagement extends connect(getStore()
...
     if (!isJsonStrMatch(this.permissions, newPermissions)) {
       this.permissions = newPermissions;
     }
 
-    this.data = getProgrammeData();
+    this.formattedData = this.formatData(this.data);
+  }
+
+  formatData = (data) => {
+    return [
```
Comment (acory) [inline intervention-results/effective-efficient-programme-mgmt/effective-efficient-programme-mgmt.ts:160]: TODO- write method that actually formats data

File: intervention-results/effective-efficient-programme-mgmt/effective-efficient-programme-mgmt.ts [hunk 3]
```diff
@@ -151,32 +145,65 @@ export class EffectiveAndEfficientProgrammeManagement extends connect(getStore()
...
-  // }
-
-  // createDialog() {
-  //   this.activityDialog = document.createElement('activity-dialog') as ActivityDialog;
-  //   this.activityDialog.setAttribute('id', 'activityDialog');
-  //   this.activityDialog.toastEventSource = this;
-  //   document.querySelector('body')!.appendChild(this.activityDialog);
-  // }
+  private openActivityDialog(activity) {
+    this.createDialog();
```
Comment (acory) [inline intervention-results/effective-efficient-programme-mgmt/effective-efficient-programme-mgmt.ts:195]: dialog being created on every open dialog click, use `openDialog` from dialog.ts file

File: intervention-results/effective-efficient-programme-mgmt/activity-dialog.ts [hunk 4]
```diff
@@ -81,6 +127,10 @@ export class ActivityDialog extends connect(getStore())(ComponentBaseMixin(LitEl
     super.connectedCallback();
   }
 
+  stateChanged(state: RootState) {
+    this.interventionId = get(state, 'app.routeDetails.params.interventionId');
```
Comment (acory) [inline intervention-results/effective-efficient-programme-mgmt/activity-dialog.ts:131]: send required data from parent, best to not connect modals to redux  (especially when using helper method openDialog from dialog.ts file)

File: intervention-results/effective-efficient-programme-mgmt/activity-dialog.ts [hunk 5]
```diff
@@ -38,24 +47,54 @@ export class ActivityDialog extends connect(getStore())(ComponentBaseMixin(LitEl
...
+            </etools-currency-amount-input>
           </div>
           <div class="col col-6">
-            <etools-currency-amount-input id="partnerContribution" label="Partner contribution">
+            <etools-currency-amount-input
+              id="partnerContribution"
+              label="Partner contribution"
+              .value="${this.activity.partner_contribution}"
+              @value-changed="${({detail}: CustomEvent) => this.updateField('partner_contribution', detail.value)}"
+              ?readonly="${this.isReadonly(this.editMode, this._permissionObj.edit.programme_management_activity)}"
```
Comment (acory) [inline intervention-results/effective-efficient-programme-mgmt/activity-dialog.ts:96]: no need for permissions checking at this stage, the user will not have access to the dialog (permissions are handled in effective-efficient-programme-mgmt.ts)

**Review by lajoskovago — APPROVED** [review]

File: intervention-results/effective-efficient-programme-mgmt/activity-dialog.ts [hunk 6]
```diff
@@ -38,62 +42,93 @@ export class ActivityDialog extends connect(getStore())(ComponentBaseMixin(LitEl
...
-        dialog-title="Add activity"
+        keep-dialog-open
+        dialog-title="Edit activity"
+        ok-btn-text="Save"
         ?opened="${this.dialogOpened}"
-        @close="${() => this.closeDialog()}"
+        @close="${() => this.onClose()}"
+        @confirm-btn-clicked="${this.onSaveClick}"
       >
+        <etools-loading ?active="${this.loadingInProcess}" loading-text="Loading..."></etools-loading>
```
Comment (acory) [inline intervention-results/effective-efficient-programme-mgmt/activity-dialog.ts:52]: Check if this etools-loading is really needed, etools-dialog has loading behavior incorporated

**Pull request merged**